src/worker_window.js

```diff
@@ -0,0 +1,24 @@
+const { app, BrowserWindow } = require('electron');
+const springPlatform = require('./spring_platform');
+
+let workerWindow;
+
+// Create a window for the worker
+
+app.prependListener('ready', () => {
+	workerWindow = new BrowserWindow({
+		show: false,
+		webPreferences: {nodeIntegration: true }
+	});
+	workerWindow.loadFile(`${__dirname}/worker_index.html`);
+
+	// Send a 'start-indexing-replays' request once the worker is ready
+	workerWindow.once('ready-to-show', () => {
+		workerWindow.send('start-indexing-replays', springPlatform.writePath);
+	});
+
+});
+
+module.exports = {
+	workerWindow: workerWindow,
```

Comment:
would this really work? I expect it to be null - forever, the assignment at line 9 comes late, and those two things just point at different things

maybe it's necessary to do something like launcher_gui's
```
	getMainWindow() {
		return mainWindow;
	}
```

Reply:
I went the function way